src/screens/Asset/Asset.js

```diff
@@ -292,6 +297,61 @@ class AssetScreen extends React.Component<Props, State> {
     }
   }
 
+  getModalActionOptions = () => {
+    const {
+      navigation,
+      baseFiatCurrency,
+      rewardActive,
+      goToInvitationFlow,
+      balances,
+      exchangeSupportedAssets,
+    } = this.props;
+    const { assetData } = navigation.state.params;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+    const { token } = assetData;
+    const balance = getBalance(balances, token);
+    const isSupportedByExchange = exchangeSupportedAssets.some(({ symbol }) => symbol === token);
+    return [
+      {
+        key: 'buy',
+        label: `Buy with a card${Platform.OS === 'ios' ? ' or Apple Pay' : ''}`,
+        iconName: 'wallet',
+        onPress: () => this.goToExchangeFlow(fiatCurrency, token),
```

Comment:
Maybe it should also be check if the asset is supported by the exchange (as you did for 'Exchange' option)?

Reply:
I don't know if these fiat-crypto assets are in the same group since they rely on external fiat providers